feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -445,12 +453,8 @@ func TestSoftwareVersion(t *testing.T) {
 func TestCPU(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
 
-	r := regexp.MustCompile("^CPU")
-	cpus := findMatchedComponents(t, dut, r)
-	t.Logf("Found CPU list: %v", cpus)
-	if len(cpus) == 0 {
-		t.Fatalf("Get CPU list for %q: got 0, want > 0", dut.Model())
-	}
+	cpus := components.FindComponentsByType(t, dut, cpuType)
```

Comment:
Also, should we keep the check for cpus length here? It seems right now if there is some error and 0 CPU found, this test will pass w/o actually testing anything (which is not ideal)

Reply:
Sim. applicable for all len check in other subtests